currency_rate_update/models/res_currency_rate_provider.py

```diff
@@ -0,0 +1,313 @@
+# Copyright 2009-2016 Camptocamp
+# Copyright 2010 Akretion
+# Copyright 2019 Brainbean Apps (https://brainbeanapps.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+import logging
+from datetime import datetime, time
+from dateutil.relativedelta import relativedelta
+
+from odoo import models, fields, api, _
+from odoo.exceptions import UserError
+
+_logger = logging.getLogger(__name__)
+
+
+class ResCurrencyRateProvider(models.Model):
+    _name = 'res.currency.rate.provider'
+    _description = 'Currency Rates Provider'
+    _inherit = ['mail.thread']
+    _rec_name = 'name'
+    _order = 'name'
+
+    company_id = fields.Many2one(
+        string='Company',
+        comodel_name='res.company',
+        required=True,
+        default=lambda self: self._default_company_id(),
+    )
+    active = fields.Boolean(
+        default=True,
+        track_visibility='onchange',
+    )
+    service = fields.Selection(
+        string='Source Service',
+        selection=lambda self: self._selection_service(),
```

Comment:
No need of a selection method. Use `selection_add` instead on inherited models. This way, we can see in the model definition the available options.